lib/src/controller/click_handler.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:link_five/src/logic/actions/move_tile_action.dart';
+import 'package:link_five/src/logic/actions/place_tile_action.dart';
+import 'package:link_five/src/logic/game_action.dart';
+import 'package:link_five/src/model/game/game_state.dart';
+import 'package:link_five/src/model/game/player_color.dart';
+import 'package:link_five/src/model/game/tile_location.dart';
+
+class ClickHandler {
+  TileLocation? source;
```

Comment:
Source should probably be private.
```suggestion
  TileLocation? _source;
```

Reply:
You can make accessible via a getter when you need it.
```dart
TileLocation? get source => _source
```